cmd/kubeadm/app/cmd/token_test.go

```diff
@@ -142,3 +195,335 @@ func TestRunCreateToken(t *testing.T) {
 		}
 	}
 }
+
+func TestNewCmdTokenGenerate(t *testing.T) {
+	var buf bytes.Buffer
+	args := []string{}
+
+	cmd := NewCmdTokenGenerate(&buf)
+	cmd.SetArgs(args)
+
+	if err := cmd.Execute(); err != nil {
+		t.Errorf("Cannot execute token command: %v", err)
+	}
+}
+
+func TestNewCmdToken(t *testing.T) {
+	var buf, bufErr bytes.Buffer
+	testConfigFile := "test-config-file"
+
+	tmpDir, err := ioutil.TempDir("", "kubeadm-token-test")
+	if err != nil {
+		t.Errorf("Unable to create temporary directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	fullPath := filepath.Join(tmpDir, testConfigFile)
+
+	f, err := os.Create(fullPath)
+	if err != nil {
+		t.Errorf("Unable to create test file %q: %v", fullPath, err)
+	}
+	defer f.Close()
+
+	testCases := []struct {
+		name          string
+		args          []string
+		configToWrite string
+		expectedError bool
+	}{
+		{
+			name:          "valid: generate",
+			args:          []string{"generate"},
+			configToWrite: "",
+			expectedError: false,
+		},
+		{
+			name:          "valid: delete",
+			args:          []string{"delete", "abcdef.1234567890123456", "--dry-run", "--kubeconfig=" + fullPath},
+			configToWrite: TestConfig,
+			expectedError: false,
+		},
+	}
+
+	cmd := NewCmdToken(&buf, &bufErr)
+	for _, tc := range testCases {
+		if _, err = f.WriteString(tc.configToWrite); err != nil {
+			t.Errorf("Unable to write test file %q: %v", fullPath, err)
+		}
+		cmd.SetArgs(tc.args)
+		err := cmd.Execute()
+		if (err != nil) != tc.expectedError {
+			t.Errorf("Test case %q: NewCmdToken expected error: %v, saw: %v", tc.name, tc.expectedError, (err != nil))
+		}
+	}
+}
+
+func TestGetSecretString(t *testing.T) {
+	secret := v1.Secret{}
+	key := "test-key"
+	if str := getSecretString(&secret, key); str != "" {
+		t.Errorf("getSecretString() did not return empty string for a nil v1.Secret.Data")
+	}
+	secret.Data = make(map[string][]byte)
+	if str := getSecretString(&secret, key); str != "" {
+		t.Errorf("getSecretString() did not return empty string for missing v1.Secret.Data key")
+	}
+	secret.Data[key] = []byte("test-value")
+	if str := getSecretString(&secret, key); str == "" {
+		t.Errorf("getSecretString() failed for a valid v1.Secret.Data key")
+	}
+}
+
+func TestGetClientset(t *testing.T) {
+	testConfigFile := "test-config-file"
+
+	tmpDir, err := ioutil.TempDir("", "kubeadm-token-test")
+	if err != nil {
+		t.Errorf("Unable to create temporary directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	fullPath := filepath.Join(tmpDir, testConfigFile)
+
+	// test dryRun = false on a non-exisiting file
+	getClientset(fullPath, false)
+
+	// test dryRun = true on a non-exisiting file
+	getClientset(fullPath, true)
```

Comment:
same here, check results ?

Reply:
@kad updated with the two requests above.

Reply:
Tests seems to be failing still:
```
	token_test.go:194: Test case invalid: print join command: RunCreateToken expected error: true, saw: false
```

Reply:
should be fixed now.